src/modules/hip/kernel/normalize.hpp

```diff
@@ -0,0 +1,1925 @@
+
+        if(hipThreadIdx_x == 0)
+            meanTensor[dstIdx] = partialSum_smem[0] / static_cast<float>(lengthX * lengthZ);
+    }
+    // compute mean along x-y-z direction
+    else if(axisMask == 7)
+    {
+        id_x *= 8;
+        __shared__ float partialSum_smem[16][16];                                        // 16 rows of src, 128 reduced cols of src in a 16 x 16 thread block
+        float *partialSumRowPtr_smem = &partialSum_smem[hipThreadIdx_y][0];              // float pointer to beginning of each row in Shared
+        partialSumRowPtr_smem[hipThreadIdx_x] = 0.0f;                                    // initialization of Shared to 0.0f using all 16 x 16 threads
+
+        uint x_index = id_x % srcStridesNZY.z;
+        uint y_index = id_x / srcStridesNZY.z;
+
+        roi = &roiTensor[id_z * 6 + 3];
+        lengthZ = roi[0];
+        lengthY = roi[1];
+        lengthX = roi[2];
+        if ((x_index >= lengthX) || (y_index >= lengthY) || (id_y >= lengthZ))
+        {
+            return;
+        }
+
+        int xAlignedLength =  lengthX & ~7;       // alignedLength for vectorized global loads
+        int xDiff = lengthX - xAlignedLength;    // difference between roiWidth and alignedLength
+        uint srcIdx = (id_z * srcStridesNZY.x) + (id_y * srcStridesNZY.y) + id_x;
+
+        d_float8 src_f8;
+        rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &src_f8);           // load 8 pixels to local memory
+        if (x_index + 8 > lengthX)
+        {
+            for(int i = xDiff; i < 8; i++)
+                src_f8.f1[i] = 0.0f;                                            // local memory reset of invalid values (from the vectorized global load) to 0.0f
+        }
+        src_f8.f4[0] += src_f8.f4[1];
+        partialSumRowPtr_smem[hipThreadIdx_x] = (src_f8.f1[0] +
+                                                 src_f8.f1[1] +
+                                                 src_f8.f1[2] +
+                                                 src_f8.f1[3]);
+        __syncthreads();
+
+        // Reduction of 16 floats on 16 threads per block in x dimension (for every y dimension)
+        reduction_sum_x_hip(partialSumRowPtr_smem);
+
+        if (hipThreadIdx_x == 0)
+        {
+            // Reduction of 16 floats on 16 threads per block in y dimension
+            for (int threadMax = 8, increment = 128; threadMax >= 1; threadMax /= 2, increment /= 2)
+            {
+                if (hipThreadIdx_y < threadMax)
+                    partialSumRowPtr_smem[0] += partialSumRowPtr_smem[increment];
+                __syncthreads();
+            }
+
+            // Final store to dst
+            if (hipThreadIdx_y == 0)
+            {
+                uint dstIdx = (id_z * hipGridDim_y * hipGridDim_x) + (hipBlockIdx_y * hipGridDim_x + hipBlockIdx_x);
+                partialSumTensor[dstIdx] = partialSumRowPtr_smem[0];
+            }
+        }
+    }
+    else if(axisMask == 6)
```

Comment:
After making changes in the test suite comments for time, please run a simple experiment for GPU and measure rocprof kernel times as well as overall wall time.
Pick 2 or 3 variants among these, say axisMask=6 and 7, and run a the current version as is.
Compare those times against a simple version where we have completely different kernels for axisMask = 6 and 7.
So yes, that will increase number of kernels, but we need to see at what cost/benefit in performance. (Since separating kernels was good for box filter)